src/stateManager/ValidationService.ts

```diff
@@ -318,6 +349,82 @@ export default class ValidationService {
 
         return signer === block.author;
     }
+
+    /* Returns the latest block timestamp for a given fork */
+    private getLatestBlockTimestamp(forkId: ForkId): Timestamp {
+        // Get the genesis timestamp from the genesis snapshot
+        const genesisSnapshot =
+            this.storage.stateSnapshots.getGenesisSnapshotDataByForkId(forkId);
+        const genesisTimestamp = genesisSnapshot.snapshot.timestamp;
+        // Get all blocks for the fork in descending order
+        const blocksIterator = this.storage.blocks.getBlocksByForkId(
```

Comment:
Imo better to rename `getBlocksByForkId` -> `getBlocksIterator` or just `getIterator`